mbedtls: update to 3.6.3 in 3.7 branch

From https://github.com/zephyrproject-rtos/zephyr/pull/87946#issuecomment-2768963833:
Mbed TLS 3.6 is LTS, Zephyr 3.7 is LTS, Zephyr 3.7.1 had Mbed TLS 3.6.2 so Zephyr 3.7.2 should have Mbed TLS 3.6.3 (fix version + 1) :)
plus Mbed TLS 3.6.3 fixes two CVEs, see https://github.com/Mbed-TLS/mbedtls/releases/tag/mbedtls-3.6.3

Hi,

I noticed build failure on my application after the update while building TF-M:

```
.../modules/crypto/mbedtls/library/md.c:251: undefined reference to `psa_can_do_hash'
```

TF-M 2.1.x upstream branch (https://git.trustedfirmware.org/plugins/gitiles/TF-M/trusted-firmware-m.git/+log/refs/heads/release/2.1.x/) has a fix https://git.trustedfirmware.org/plugins/gitiles/TF-M/trusted-firmware-m.git/+/bdffde9c634c9277c128e19510653be3c136d1a2 which I believe should at least be taken into Zephyr's 3.7 TF-M branch. With the change included, I can get TF-M to pass the build.

There at least two mbedTLS update related changes as well. So maybe it makes most sense to update the whole TF-M to the latest 2.1.x version?

--- a/doc/releases/release-notes-3.7.rst
+++ b/doc/releases/release-notes-3.7.rst
@@ -38,6 +38,15 @@ https://github.com/Mbed-TLS/mbedtls/releases/tag/mbedtls-3.6.3
 Mbed TLS 3.6 is an LTS release that will be supported
 with security and bug fixes until at least March 2027.
 
+Trusted Firmware-M (TF-M)
+*************************
+
+TF-M was updated to version 2.1.2 (from 2.1.1). The release notes can be found at:
+https://trustedfirmware-m.readthedocs.io/en/tf-mv2.1.2/releases/2.1.2.html
+
+TF-M 2.1 is an LTS release that will be supported
+with security and bug fixes until at least May 2027.
+
 .. _zephyr_3.7.1:
 
 Zephyr 3.7.1

--- a/submanifests/optional.yaml
+++ b/submanifests/optional.yaml
@@ -40,7 +40,7 @@ manifest:
       groups:
         - optional
     - name: tf-m-tests
-      revision: 502ea90105ee18f20c78f710e2ba2ded0fc0756e
+      revision: deeb187ef8cc3378f44d940f9d5733b07b90b277
       path: modules/tee/tf-m/tf-m-tests
       remote: upstream
       groups:

--- a/west.yml
+++ b/west.yml
@@ -327,7 +327,7 @@ manifest:
       groups:
         - crypto
     - name: trusted-firmware-m
-      revision: 60ebade5d3d381a210af90191e475d8870b8adbc
+      revision: 78854fffd4f57d91547dc3e15370d786ead01477
       path: modules/tee/tf-m/trusted-firmware-m
       groups:
         - tee

